### DISCUSSION TITLE
Updating beatmaps should preserve custom difficulties

### DISCUSSION THREAD

**Discussion by Doksha** [ppy/osu#20607] [Ideas]
In stable if you update a beatmap it will preserve custom difficulties, but if you misclick the update button in lazer you lose all your difficulties and however many hours of replays you had on those difficulties (this happened to me, losing around 10 hours of practice replays on a map because I misclicked in a multi lobby) 
Currently the system checks if all maps in the *local* mapset are reflected in the online mapset, when it should be the other way around; the system checking if all maps in the *online* mapset are reflected locally, and only adding maps to the local mapset, not removing any.

**Comment by peppy** [comment]
Any local beatmaps should be left intact in another set. There's also an undelete option in settings. This is all considered with the implementation.

Please attach your database, and if possible a video showing what you imply is happening.

**Reply by Doksha** [reply]
This does not occur in beatmaps made in lazer, but since no sane person would use the lazer editor in the current state, basically every custom difficulty is seen as needing to be updated (as its imported from stable), and will not be sent to a separate set.
I was unaware of the restore recently deleted beatmaps button sadly, yes it does work, but the other issue should still be fixed.

**Reply by peppy** [reply]
> but since no sane person would use the lazer editor in the current state

Out of curiosity alone, could you explain why the editor is unusable for the purpose of making "custom difficulties"? That should be a cut and dry simple use case. I've tested and it worked fine and took only a few seconds.

**Reply by Doksha** [reply]
> Out of curiosity alone, could you explain why the editor is unusable for the purpose of making "custom difficulties"? That should be a cut and dry simple use case. I've tested and it worked fine and took only a few seconds.


I mean, the issue with hit circles not selecting unless you go move slowly is easily the biggest reason. Additionally, creating a custom difficulty without deleting all hit objects duplicates the *previously saved* difficulty, not the hit objects that are currently in ram. This makes creating practice difficulties extremely tedious as you have to open the main difficulty *every single time*.
I also liked having a simple popup menu when I clicked song setup, instead of having an entirely new menu which is harder to navigate.
The stable editor is quite well designed, the lazer editor seems to be trying to hard to be something else when it doesnt need to be.
Additionally, afaik, you dont have access to the actual files, so screw anyone who prefers that way of editing. (this also applies to skinning, which is near impossible in lazer unless there is some feature I am unaware of)

In terms of what probably 90% of people are going to use the editor for, the lazer editor is insufficient.

**Reply by peppy** [reply]
> I mean, the issue with hit circles not selecting unless you go move slowly is easily the biggest reason. 

This is already fixed.

> Additionally, creating a custom difficulty without deleting all hit objects duplicates the *previously saved* difficulty, not the hit objects that are currently in ram. This makes creating practice difficulties extremely tedious as you have to open the main difficulty *every single time*.

Could you open an issue?

> I also liked having a simple popup menu when I clicked song setup, instead of having an entirely new menu which is harder to navigate.

Can you better describe what is feeling "not simple" in a separate discussion?


> The stable editor is quite well designed, the lazer editor seems to be trying to hard to be something else when it doesnt need to be.

You will likely find this is partly due to being used to something.

**Reply by Doksha** [reply]
> This is already fixed.

I tested this before posting and just now, I have no idea what you think is fixed, but it isnt this
(opening other discussions for the other issues as requested)

**Reply by peppy** [reply]
Please open a new discussion showing the case you believe is not fixed.

**Reply by Doksha** [reply]
#16410
you have been recently active in this discussion (in the pull request) so I know you know about it, I dont think you actually want me to create a new discussion. (maybe I just articulated myself badly and you misinterpreted what the issue was)

**Reply by peppy** [reply]
Right, I thought that was already in. Expect the fix next release!
